common/djangoapps/student/migrations/0001_squashed_0031_auto_20200317_1122.py

```diff
@@ -88,7 +89,7 @@ class Migration(migrations.Migration):
             name='CourseEnrollment',
             fields=[
                 ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
-                ('course_id', opaque_keys.edx.django.models.CourseKeyField(db_index=True, max_length=255)),
+                ('course', models.ForeignKey(db_constraint=False, on_delete=django.db.models.deletion.CASCADE, to='course_overviews.CourseOverview')),
```

Comment:
1. I still wonder if this `db_constraint=False` was meant to be temporary, but as-is, I think it is correct to have this.
2. Aren't you missing details from `opaque_keys.edx.django.models.CourseKeyField(db_column='course_id', db_index=True, max_length=255),`?  If you delete ALL migration files here and run migrate, what does it create for the initial migration?  Maybe it isn't worth editing this stuff at all?  I don't know how much this buys us and whether it is worth the risk?

Reply:
2. I did try to rebuild from scratch, but there were some dependency issues that caused problems when I tried it. When I switched this over, I should have also removed the `AlterField` that had the more complete `CourseKeyField` arguments.

Reply:
Removed?  You did remove it, but, don't you want to use it?  For example, aren't you now missing the `db_column='course_id'`, as just one example?

Reply:
Also, I'm still not clear where and if we gain any improvements from this squashing, and whether it is simpler/better to just go with the auto-generated in this case to not be concerned with introducing bugs, especially because you couldn't re-generate the initial migration.

Reply:
This is from the read-replica.  Does your local match this?:
```
mysql> describe student_courseenrollment;
+-----------+--------------+------+-----+---------+----------------+
| Field     | Type         | Null | Key | Default | Extra          |
+-----------+--------------+------+-----+---------+----------------+
| id        | int(11)      | NO   | PRI | NULL    | auto_increment |
| user_id   | int(11)      | NO   | MUL | NULL    |                |
| course_id | varchar(255) | NO   | MUL | NULL    |                |
| created   | datetime     | YES  | MUL | NULL    |                |
| is_active | tinyint(1)   | NO   |     | NULL    |                |
| mode      | varchar(100) | NO   |     | NULL    |                |
+-----------+--------------+------+-----+---------+----------------+
6 rows in set (0.00 sec)
```

Reply:
Different order, but it does seem to match.

```
mysql> describe student_courseenrollment;
+-----------+--------------+------+-----+---------+----------------+
| Field     | Type         | Null | Key | Default | Extra          |
+-----------+--------------+------+-----+---------+----------------+
| id        | int(11)      | NO   | PRI | NULL    | auto_increment |
| course_id | varchar(255) | NO   | MUL | NULL    |                |
| created   | datetime(6)  | YES  | MUL | NULL    |                |
| is_active | tinyint(1)   | NO   |     | NULL    |                |
| mode      | varchar(100) | NO   |     | NULL    |                |
| user_id   | int(11)      | NO   | MUL | NULL    |                |
+-----------+--------------+------+-----+---------+----------------+
6 rows in set (0.08 sec)
```

Reply:
I see.  I guess that is just how ForeignKey works its magic?

Reply:
I imagine so. There really is a lot of magic going on.